From 1dd8df1a030c4ef922e2743220131a001749f8b9 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Fri, 22 Nov 2024 16:52:33 +0100
Subject: [PATCH 1461/1695] plat-stm32mp1: scmi_server: MP15 denies GPIOZ reset
 domain

Forbid access to GPIOZ reset controller through the SCMI GPIOZ reset
domain. It is still exposed but non-secure world is not allowed to use it.

This change was squashed in upstream change [1].

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 10a0ab75c61ed1c52fc4a4f570f4fb00dc4f9784 plat-stm32mp1: scmi_server: MP15 denies GPIOZ reset domain]
Change-Id: I8c2644e30555014138a8c365de2a5439a19b99a9
Link: https://github.com/OP-TEE/optee_os/commit/b18ace9b2cebb491aa3a2b7544a2f16b9a0b41bb [1]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/420414
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/scmi_server.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/scmi_server.c b/core/arch/arm/plat-stm32mp1/scmi_server.c
index c6c01eedd..05fdb9bb6 100644
--- a/core/arch/arm/plat-stm32mp1/scmi_server.c
+++ b/core/arch/arm/plat-stm32mp1/scmi_server.c
@@ -214,7 +214,7 @@ static struct stm32_scmi_rd stm32_scmi_reset_domain[] = {
 	RESET_CELL_DECPROT(RST_SCMI_STGEN, STGEN_R, "stgen",
 			   STM32MP1_ETZPC_STGENC_ID),
 	RESET_CELL_DECPROT(RST_SCMI_GPIOZ, GPIOZ_R, "gpioz",
-			   STM32MP1_ETZPC_GPIOZ_ID),
+			   ETZPC_ID_NEVER_ACCESSIBLE),
 	RESET_CELL_DECPROT(RST_SCMI_CRYP1, CRYP1_R, "cryp1",
 			   STM32MP1_ETZPC_CRYP1_ID),
 	RESET_CELL_DECPROT(RST_SCMI_HASH1, HASH1_R, "hash1",
-- 
2.47.2

